src/shared/sam/cli/samCliConfiguration.ts

```diff
@@ -55,8 +55,12 @@ export class DefaultSamCliConfiguration implements SamCliConfiguration {
         const detectedLocation = (await this._samCliLocationProvider.getLocation()) ?? ''
         // Avoid setting the value redundantly (could cause a loop because we
         // listen to the `onDidChangeConfiguration` event).
-        if (configLocation !== detectedLocation) {
-            await this.setSamCliLocation(detectedLocation)
+        if (detectedLocation && configLocation !== detectedLocation) {
```

Comment:
If SAM CLI is no longer detected, why aren't we writing this to the config? Seems like the toolkit would incorrectly think the CLI is at some location.

Reply:
I don't see any callsites where that would matter. But https://github.com/aws/aws-toolkit-vscode/pull/1242 is a more complete solution that updates all call sites to explicitly deal with just-in-time autodetection instead of doing contortions to update the user's config (which is not actually necessary and a bit of an anti-pattern).

Reply:
It could be misleading in that you've run the "detect sam cli" command, and the toolkit shows the location configured as the old path (instead of no path). I'm fine proceeding with this in anticipation of 1242 then.